src/tests/test_schedulers.cpp

```diff
@@ -357,6 +384,176 @@ SCENARIO("New thread scheduler depends on subscription")
+                });
+
+                call_stack.emplace_back("task 2 ends");
+                return std::nullopt;
+            });
+
+            call_stack.emplace_back("task 1 ends");
+            return std::nullopt;
+        });
+
+        THEN("order of call-stack must be in order")
+        {
+            REQUIRE(call_stack == std::vector<std::string>{
+                "task 1 starts",
+                "task 1 ends",
+                "task 2 starts",
+                "task 2 ends",
+                "task 3 runs",
+            });
+        }
+    }
+}
+
+SCENARIO("trampoline scheduler is thread local")
+{
+    WHEN("two threads are using the same trampoline scheduler")
+    {
+        auto scheduler = rpp::schedulers::trampoline{};
+        auto sub       = rpp::composite_subscription{};
+        auto worker    = scheduler.create_worker(sub);
+        rpp::subscription_guard guard{sub};
+
+        std::promise<std::vector<std::thread::id>> call_stack_1;
+        std::promise<std::vector<std::thread::id>> call_stack_2;
+        auto stack_future_1 = call_stack_1.get_future();
+        auto stack_future_2 = call_stack_2.get_future();
+
+        auto thread_id_1 = simulate_nested_scheduling(worker, call_stack_1);
+        auto thread_id_2 = simulate_nested_scheduling(worker, call_stack_2);
+
+        THEN("call stack of two threads shall be separate")
+        {
+            REQUIRE(stack_future_1.wait_for(std::chrono::seconds{5}) == std::future_status::ready);
+            REQUIRE(stack_future_2.wait_for(std::chrono::seconds{5}) == std::future_status::ready);
+
+            REQUIRE(stack_future_1.valid());
+            REQUIRE(stack_future_2.valid());
+
+            REQUIRE(stack_future_1.get() == std::vector{thread_id_1, thread_id_1, thread_id_1});
+            REQUIRE(stack_future_2.get() == std::vector{thread_id_2, thread_id_2, thread_id_2});
+        }
+    }
+}
+
+SCENARIO("trampoline scheduler regards unsubscribed subscription")
+{
+    GIVEN("unsubscribed subscription")
+    {
+        auto scheduler = rpp::schedulers::trampoline{};
+        auto sub       = rpp::composite_subscription{};
+        auto worker    = scheduler.create_worker(sub);
+
+        sub.unsubscribe();
+
+        THEN("shall not see execution of schedulable")
+        {
+            worker.schedule([&]() -> rpp::schedulers::optional_duration
+            {
+                REQUIRE(false);
+                return std::nullopt;
+            });
+        }
+    }
+
+    GIVEN("asynchronously unsubscribes subscription")
+    {
+        auto scheduler = rpp::schedulers::trampoline{};
+        auto sub       = rpp::composite_subscription{};
+        auto worker    = scheduler.create_worker(sub);
+
+        THEN("shall not see execution of inner schedulable")
+        {
+            std::jthread async_stopper([=] {
+                std::this_thread::sleep_for(std::chrono::seconds{1});
+                sub.unsubscribe();
+            });
```

Comment:
Can't see any benefit of using thread there comparing to just calling unsubscribe in place of 
```cpp
// Block return so that async stopper could prevent the inner schedulable from running.
std::this_thread::sleep_for(std::chrono::seconds{3});
```

Reply:
Threads and sleeps add complexity to test, but doesn't provide any benefit and just adds race-condition.

Reply:
You're right! I've been having this concern for long time.